### DISCUSSION TITLE
Sumilation with ellipsoid

### DISCUSSION THREAD

**Discussion by Flojo1994** [woodem/woo#44] [Q&A]
Dear community,

I have tried to simulate falling ellipsoid of a vibration plate. The plate is an import as a stl-file. I read in the documentation that the interaction between ellipsoids and faces does not work. I am interested for a creative solution. 

Thank you for you help.

Best regards 

Florian

**Comment by eudoxos** [comment]
Hi Florian,

there was an attempt at implementation in https://github.com/woodem/woo/blob/f1699890b9c77839ac8a3de614284efe46237eb2/src/dem/Ellipsoid.cpp#L133 but as you see in the code, that part was never finished. If you feel like it and want to apply your knowledge of numerics & geometry and learn stuff on the way, you are welcome to crack it — the issue is basically computing the closest points between ellipsoid and facet (plane, lines, points)... I did not have success back then, and not time to get back to that. Perhaps you find a paper which will describe some algorithm in detail, perhaps it is implemented somewhere already and could be adapted; perhaps it needs to be iterative, in which case the Ellipsoid-Ellipsoid contact might be of inspiration (in the same file).

Beside that, you can think of replacing your ellipsoids with capsules (they are geometrically much simpler, and often do the job just fine).

HTH,

Vaclav

**Comment by eudoxos** [comment]
PS of course if you need some help while working on that, just write a message here.
